src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -7420,8 +7420,14 @@ Strings MergeTreeData::getDataPaths() const
 }
 
 
-void MergeTreeData::reportBrokenPart(MergeTreeData::DataPartPtr & data_part) const
+void MergeTreeData::reportBrokenPart(MergeTreeData::DataPartPtr data_part) const
 {
+    if (!data_part)
+        return;
```

Comment:
Why do we report `nullptr` as a broken part? (I would prefer to move this `if` to `DataPartsExchange`, to the `catch` block)